src/containers/Firmware/index.js

```diff
@@ -0,0 +1,377 @@
+import Button from 'components/Button';
+import Modal from 'components/Modal';
+import styles from './index.module.scss';
+import AssignmentModal from './components/AssignmentModal';
+import VersionModal from './components/VersionModal';
+
+const Firmware = ({
+  firmwareData,
+  trackAssignmentData,
+  onDeleteTrackAssignment,
+  onDeleteFirmware,
+  onCreateFirnware,
+  onUpdateFirmware,
+}) => {
+  const [addAssignmentModal, setAddAssignmentModal] = useState(false);
+  const [editAssignmentModal, setEditAssignmentModal] = useState(false);
+  const [addVersionModal, setAddVersionModal] = useState(false);
+  const [editVersionModal, setEditVersionModal] = useState(false);
+  const [deleteAssignmentModal, setDeleteAssignmentModal] = useState(false);
+  const [deleteVersionModal, setDeleteVersionModal] = useState(false);
+  const [taskAssignmentValues, setTaskAssignmentValues] = useState({
+    firmwareTrackId: 0,
+    firmwareVersionId: 0,
+  });
+
+  const [firmwareValues, setFirmwareValues] = useState({
+    id: 0,
+    modelId: '',
+    versionName: '',
+    description: '',
+    commit: '',
+    releaseDate: '',
+    filename: '',
+    validationCode: '',
+    createdTimestamp: '',
+    lastModifiedTimestamp: '',
+  });
+
+  const deleteTrackAssignment = () => {
+    const { firmwareTrackId, firmwareVersionId } = taskAssignmentValues;
+    onDeleteTrackAssignment(firmwareTrackId, firmwareVersionId);
+    setDeleteAssignmentModal(false);
+  };
+  const deleteFirmware = () => {
+    const { id } = firmwareValues;
+    onDeleteFirmware(id);
+    setDeleteVersionModal(false);
+  };
+  const createFirmware = ({
+    modelId,
+    versionName,
+    description,
+    filename,
+    commit,
+    date,
+    validationCode,
+  }) => {
+    if (date !== null) {
+      const releaseDate = date.valueOf().toString();
+      onCreateFirnware(
+        modelId,
+        versionName,
+        description,
+        filename,
+        commit,
+        releaseDate,
+        validationCode
+      );
+    } else {
+      onCreateFirnware(modelId, versionName, description, filename, commit, validationCode);
+    }
+    setAddVersionModal(false);
+  };
+
+  const updateFirmware = ({
+    modelId,
+    versionName,
+    description,
+    filename,
+    commit,
+    date,
+    validationCode,
+  }) => {
+    const { id, createdTimestamp, lastModifiedTimestamp } = firmwareValues;
+    if (date !== null) {
+      const releaseDate = date.valueOf().toString();
+      onUpdateFirmware(
+        id,
+        modelId,
+        versionName,
+        description,
+        filename,
+        commit,
+        releaseDate,
+        validationCode,
+        createdTimestamp,
+        lastModifiedTimestamp
+      );
+    } else {
+      onUpdateFirmware(
+        id,
+        modelId,
+        versionName,
+        description,
+        filename,
+        commit,
+        validationCode,
+        createdTimestamp,
+        lastModifiedTimestamp
```

Comment:
this doesn't work. your number of params is different, so the else statement is going to set validationCode as releaseDate.

remove the if statement and just do:
```
 onUpdateFirmware(
        id,
        modelId,
        versionName,
        description,
        filename,
        commit,
       date ? date.valueOf().toString() : null,
        validationCode,
        createdTimestamp,
        lastModifiedTimestamp
      );
```